packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts

```diff
@@ -54,4 +56,9 @@ export class TelemetryUtils {
       "error-message": errorMessage,
     };
   };
+
+  private static addAppIdInProperty(properties:{ [key: string]: string }, ctx: PluginContext): void {
+    const appId = ctx.configOfOtherPlugins.get(Plugins.solution)?.get(ConfigKeysOfOtherPlugin.remoteTeamsAppId);
+    properties[Telemetry.appId] = appId as string;
```

Comment:
undefined as string may throw exception?

Reply:
Here it will not throw exception in my test. And also confirmed with zhaofeng in his pr.

Reply:
Better no to do like this. It is not best practice I think.